icu4j/releases_tools/api_reports.sh

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bash
+# Copyright (C) 2023 and later: Unicode, Inc. and others.
+# License & terms of use: http://www.unicode.org/copyright.html
+
+export MAVEN_ARGS='--no-transfer-progress'
+
+export artifact_version='74.0.1-SNAPSHOT'
```

Comment:
I assume the hardcoded version here is a temporary thing. `74.0.1-SNAPSHOT` is also referred by line 39 in this shell script. Ideally, we want to pass version to this shell script, instead of embedding it. If it is too much, then I'd prefer to define it in one place.

Reply:
Everything in "release_tools" is work in progress.
Just making sure I have things in one place and don't get lost, instead of just having them on my machine.
Will fine-tune them once all the ant tasks have a maven equivalent.

Reply:
TLDR: this works as expected.

The way "gatherapi" works is that `com.ibm.icu.dev.tool.docs.GatherAPIData` is a docket.
It completely replaces the standard `javadoc`
It is invoked for each class and method, but instead of collecting the comments and generating javadoc, it only adds the method names to the API report.

https://docs.oracle.com/javase/8/docs/technotes/guides/javadoc/doclet/overview.html

So the failure is because the `maven-javadoc-plugin` does not detect the generated `javadoc`
And it does not detect it because it does not exist, as expected (it is not generated).

This didn't fail before because the ant script invoked the javadoc generation with the custom doclet,
but didn't check that the javadoc was generated, the way maven does.

The real javadoc is generated in another maven profile, which uses the standard javadoc doclet.